server/routes/snippet.js

```diff
@@ -8,21 +8,140 @@ const router = express.Router();
 
 /////// Upload code route handler ///////
 router.post("/upload", auth, balance, async (req, res) => {
-	const snippet = new Snippet({ ...req.body, author: req.user._id });
-	try {
-		await snippet.save();
-		req.user.balance -= 1;
-		await req.user.save();
-		res.status(201).send(snippet);
-	} catch (e) {
-		res.status(400).send(e);
-	}
+  const snippet = new Snippet({ ...req.body, author: req.user._id });
+  try {
+    await snippet.save();
+    req.user.balance -= 1;
+    await req.user.save();
+    res.status(201).send(snippet);
+  } catch (e) {
+    res.status(400).send(e);
+  }
 });
 
 /////// Retrieve code route handler ///////
 router.get("/snippet", auth, async (req, res) => {
-	const snippets = await Snippet.find({ author: req.user._id });
-	res.send(snippets);
+  const snippets = await Snippet.find({ author: req.user._id });
+  res.send(snippets);
 });
 
+// fetch requested reviews
+router.get("/requested", auth, async (req, res) => {
+  try {
+    const requestedReviews = await Snippet.find({ author: req.user._id })
+      .populate("author", ["name"])
+      .populate("reviewer", ["name"]);
+    return res.status(200).json(requestedReviews);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// fetch received reviews
+router.get("/received", auth, async (req, res) => {
+  try {
+    const receivedReviews = await Snippet.find({
+      reviewer: req.user._id,
+    })
+      .populate("author", ["name"])
+      .populate("reviewer", ["name"]);
+
+    return res.status(200).json(receivedReviews);
+  } catch (error) {
+    console.error(err);
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// accept a review
+router.patch("/accept/:review_id", auth, async (req, res) => {
+  try {
+    const receivedReviews = await Snippet.find({ reviewer: req.user._id });
+
```

Comment:
I don't understand this part. If a reviewer wants to accept a specific snippet, why should we fetch all snippets they have reviewed so far? Shouldn't we just fetch that snippet based on `req.params`  and set the `reviewer._id `on the `reviewer `field of and save it?

Reply:
By doing this, I intended that only the reviewer who is assigned can accept the review, not anyone

Reply:
Hmm, I still don't quite understand :D If the user has not accepted to review yet, why whould their id exist on that snippet reviewer field? Ok, let's talk about it later, it would be easier rather than commenting back and forth :)